packages/salesforcedx-utils-vscode/src/services/telemetry.ts

```diff
@@ -383,4 +385,60 @@ export class TelemetryService {
         .catch(err => console.error(err));
     }
   }
+
+  private getRandomUserId = (): string => randomBytes(20).toString('hex');
+
+  private buildCliTelemetryCommand(): Command {
+    return new SfCommandBuilder()
+      .withArg('telemetry')
+      .withJson()
+      .build();
+  }
+
+  private async executeCliTelemetry(
+    command: Command
+  ): Promise<string> {
+    const workspacepath = workspaceUtils.getRootWorkspacePath();
+    const execution = new CliCommandExecutor(command, { cwd: workspacepath }
+    ).execute();
+    const cmdOutput = new CommandOutput();
+    const result = cmdOutput.getCmdResult(execution);
+    return result;
+  }
+
+  public async getUserId(): Promise<string> {
+    // Defining UserId in globalState and using the same in appInsights reporter.
+    // Assigns cliId to UserId when it's undefined in global state.
+    // cliId is undefined when cli-telemetry variable disable-telemetry is true.
+    let globalStateUserId = this.extensionContext?.globalState.get<
+      string | undefined
+    >('UserId');
+
+    if (globalStateUserId) {
+      return globalStateUserId;
```

Comment:
needs test case.

Reply:
added a test where the function returns telemetryUserId if defined.